### PR TITLE
[pixman] Update download

### LINKED ISSUE #28537: [pixman] update to 0.42.2
Library name: pixman

New version number: 0.42.2

Other information that may be useful (release notes, etc...)
https://github.com/freedesktop/pixman/compare/pixman-0.40.0...pixman-0.42.2

### PR DESCRIPTION
**Pull request by SchaichAlonso**
**Describe the pull request**
Update download.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  unchanged

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  `Yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: versions/p-/pixman.json [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "38e0101f0f688114057f139d0f3879c0e512cc3d",
+      "version": "0.42.2",
+      "port-version": 0
+    },
+    {
+      "git-tree": "3693f8949db091578e1f9101d2a00f90f3667740",
+      "version": "0.42.0",
+      "port-version": 0
+    },
```
Comment (Thomas1664) [inline versions/p-/pixman.json:12]:
```suggestion
```

Please add only one version.

Comment (SchaichAlonso) [reply versions/p-/pixman.json]: done

**Comment by SchaichAlonso** [comment]
It seems arm64 breaks with the update, which might get fixed by  #28024 (don't have an arm64 here to give it a try).
I can rebase / cherry on #28024 and have the CI attempt compiling 0.42.2...

File: ports/pixman/portfile.cmake [hunk 2]
```diff
@@ -29,27 +29,23 @@ elseif(VCPKG_TARGET_ARCHITECTURE MATCHES "mips")
...
 endif()
 
-set(PIXMAN_VERSION 0.40.0)
-vcpkg_download_distfile(ARCHIVE
-    URLS "https://www.cairographics.org/releases/pixman-${PIXMAN_VERSION}.tar.gz"
-    FILENAME "pixman-${PIXMAN_VERSION}.tar.gz"
-    SHA512 063776e132f5d59a6d3f94497da41d6fc1c7dca0d269149c78247f0e0d7f520a25208d908cf5e421d1564889a91da44267b12d61c0bd7934cd54261729a7de5f
-)
-vcpkg_extract_source_archive_ex(
+vcpkg_from_github(
```
Comment (MonicaLiu0311) [inline ports/pixman/portfile.cmake:32]: Duplicate: https://github.com/microsoft/vcpkg/pull/28240, Is the source of this github official?

Comment (SchaichAlonso) [reply ports/pixman/portfile.cmake:32]: I didn't realize someone else had already done this (checked the issues only rather then also checking PRs)

The repository is hosted by the developers, so I assume its' official. Not sure whether [github](https://github.com/freedesktop/pixman) or [gitlab](https://gitlab.freedesktop.org/pixman/pixman) is the primary though.

Comment (MonicaLiu0311) [reply ports/pixman/portfile.cmake:32]: > I didn't realize someone else had already done this (checked the issues only rather then also checking PR

Thanks for your contribution, wait for https://github.com/microsoft/vcpkg/pull/28240 to be merged, and then you can continue your edit based on it.

> The repository is hosted by the developers, so I assume its' official. Not sure whether [github](https://github.com/freedesktop/pixman) or [gitlab](https://gitlab.freedesktop.org/pixman/pixman) is the primary though.

Judging by lines [11-17](https://github.com/freedesktop/pixman/blob/master/README#L11-L17) and lines [43-45](https://github.com/freedesktop/pixman/blob/master/README#L43-L45) of the readme, gitlab should be the primary one.

**Comment by MonicaLiu0311** [comment]
Now #28522 and https://github.com/microsoft/vcpkg/pull/28240 have been merged, please resolve the conflict and continue your edit.

**Review by MonicaLiu0311 — APPROVED** [review]

**Comment by BillyONeal** [comment]
> It seems arm64 breaks with the update, which might get fixed by #28024 (don't have an arm64 here to give it a try). I can rebase / cherry on #28024 and have the CI attempt compiling 0.42.2...

I observe that the contents of the downloads are actually different, in particular there's an arm64 file that's different:

image

Are you sure that this is actually the same version?

**Pull request merged**